tests/test_cuda_targeted_insertion_mover.py

```diff
@@ -549,21 +585,15 @@ def test_targeted_insertion_brd4_rbfe_with_context(brd4_rbfe_state, radius, prec
         check_force_norm(-du_dx)
 
 
-@pytest.mark.memcheck
 @pytest.mark.parametrize("radius", [1.0])
-@pytest.mark.parametrize("proposals_per_move", [1, 100])
+@pytest.mark.parametrize("proposals_per_move, batch_size", [(2, 2), (100, 100), (512, 512), (2000, 1000)])
 @pytest.mark.parametrize("precision", [np.float64, np.float32])
-@pytest.mark.parametrize("seed", [2023])
-def test_tibd_exchange_deterministic_moves(radius, proposals_per_move, precision, seed):
-    """Given one water the exchange mover should accept every move and the results should be deterministic if the seed and proposals per move are the same.
-
-
-    There are three forms of determinism we require:
-    * Constructing an exchange move produces the same results every time
-    * Calling an exchange move with one proposals per move or K proposals per move produce the same state.
-      * It is difficult to test each move when there are K proposals per move so we need to know that it matches the single proposals per move case
-    * TBD: When we attempt K proposals in a batch (each proposal is made up of K proposals) it produces the same as the serial version
+@pytest.mark.parametrize("seed", [2024])
+def test_tibd_exchange_deterministic_batch_moves(radius, proposals_per_move, batch_size, precision, seed):
```

Comment:
Can we add a determinism test on the buckyball system *and* a mover that's not just limited to the first 2 waters? 

I'd like to stress-test cases where we can get a sequence of accept/rejects both within outside->inside and inside->outside. If testing with batch_size=1 is too small, we can also assert consistency between `batch_size=50 vs batch_size=250`

Reply:
Will add a test on the buckyball for determinism

Reply:
Added test for determinism of buckyball (without MD like the example tests) https://github.com/proteneer/timemachine/pull/1283/commits/2a033d61e819bd170cc1a58fe6f2a84e64f34ea5

Reply:
Would it be feasible to also add MD in as well? (Since MD should also be deterministic)

Reply:
What would be the benefit of adding MD here when we have the example tests?